Services/CO.CDP.Person.WebApi/Api/ValidationOption1.cs

```diff
@@ -0,0 +1,78 @@
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System;
+
+namespace CO.CDP.Person.WebApi.Api.ValidationSpikeOne
+{
+    public class Person
+    {
+        [Required(ErrorMessage = "ID is required.")]
+        [RegularExpression("^[0-9]+$", ErrorMessage = "ID must be numeric.")]
+        public string Id { get; init; }
+
+        [Required(ErrorMessage = "Name is required.")]
+        [StringLength(100, ErrorMessage = "Name must not exceed 100 characters.")]
+        public string Name { get; init; }
+
+        [Range(1, 120, ErrorMessage = "Age must be between 1 and 120.")]
+        public int Age { get; init; }
+
+        [Required]
+        [EmailAddress(ErrorMessage = "Email must be a valid email address.")]
+        public string Email { get; init; }
+    }
+
+    public class NewPerson
+    {
+        [Required(AllowEmptyStrings = true)]
+        public string Name { get; init; }
+        public int Age { get; init; }
+        [EmailAddress]
+        public string Email { get; init; }
+    }
+
+    [ApiController]
+    [Route("[controller]")]
+    public class PersonsController : ControllerBase
+    {
+        private static readonly Dictionary<string, Person> _persons = Enumerable.Range(1, 5)
+            .ToDictionary(index => index.ToString(), index => new Person
+            {
+                Id = index.ToString(),
+                Name = $"Sussan Tables {index}",
+                Age = 40 + index,
+                Email = "sussan@example.com"
+            });
+
+        [HttpPost]
+        public IActionResult CreatePerson([FromBody] NewPerson newPerson)
+        {
+            if (!ModelState.IsValid)
```

Comment:
I'm aware it's only an example, but I wanted to clarify how we'd do it in practice.

I think the controller's responsibility is to:

* parse the request to construct the command
* pass the command to the service or model
* interpret the result
* construct the response

Any validation done in the web layer should be related to the HTTP message (most often its body payload) to confirm we can accept the request.

Model validation should be done in layers called by the web layer. The result should then be returned back from the service or model layer in a form of a return value or an exception.